src/components/Homepage/HomePage.jsx

```diff
@@ -1,61 +1,105 @@
 import React, { useEffect } from "react";
 import { useState } from "react";
-import { Link } from "react-router-dom";
-const axios = require("axios");
+import { Link, useNavigate, useParams } from "react-router-dom";
+import axios from "axios";
+import Spinner from "../Spinner/Spinner";
 
 const api_key = process.env.REACT_APP_API_KEY;
 
-export default function HomePage({ setMovieInfo }) {
+export default function HomePage({ isLoading, setIsLoading }) {
   const [movies, setMovies] = useState([]);
+  const [totalCountPages, setTotalCountPages] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const navigate = useNavigate();
+  const params = useParams();
 
   useEffect(() => {
     fetchData();
+    fetchPageData(params.page);
   }, []);
 
-  async function fetchMovieData(id) {
-    const response = await axios.get(
-      `https://api.themoviedb.org/3/movie/${id}`,
-      {
-        params: {
-          api_key: api_key,
-        },
-      }
-    );
-    console.log(response);
-    setMovieInfo(response.data);
+  async function fetchData() {
+    setIsLoading(true);
+    try {
+      const response = await axios.get(
+        "https://api.themoviedb.org/3/movie/popular",
+        {
+          params: {
+            api_key: api_key,
+          },
+        }
+      );
+      setTotalCountPages(response.data.total_pages);
+      setMovies(response.data.results);
+      setIsLoading(false);
+    } catch (error) {
+      console.log(error);
+    }
   }
 
-  async function fetchData() {
-    const response = await axios.get(
-      "https://api.themoviedb.org/3/movie/popular",
-      {
-        params: {
-          api_key: api_key,
-        },
-      }
-    );
-    setMovies(response.data.results);
+  async function fetchPageData(page = 1) {
```

Comment:
fetchData and  fetchPageData are calling the same API route. Can we combine these 2 functions into 1?

Reply:
fixed